logstage/logstage-core/src/main/scala-2/izumi/logstage/macros/LogValuesMacro.scala

```diff
@@ -0,0 +1,31 @@
+package izumi.logstage.macros
+
+import logstage.Log
+
+import scala.annotation.tailrec
+import scala.reflect.macros.blackbox
+
+class LogValuesMacro[C <: blackbox.Context](val c: C) {
+  import c.universe.*
+  def createMessage(
+    values: Seq[c.Expr[Any]]
+  ): c.Expr[Log.Message] = {
+    val messageString = createMessageString(values)
+    c.Expr[Log.Message](q"""
+       _root_.izumi.logstage.api.Log.Message.apply($messageString)
```

Comment:
@nikita-ty754 You may directly call `LogMessageMacro` instead to improve avoid performance by avoiding a nested macro expansion

```suggestion
new LogMessageMacro0[c.type](c, false).logMessageMacro(c.Expr[String](messageString))
```